lib/grplatt.gi

```diff
@@ -3335,57 +3335,254 @@ local c,n,deg,ind,core,i,j,sum;
 
 end);
 
+
+# utility function: Find subgroup of $G$, containing $G'$ of index up to
+# limit which is normal in the normalizer of G in P.
+BindGlobal("BoundedIndexAbelianized",function(P,G,limit)
+local d,ind,i,ma,ab,a,p,b,c,e,n;
+  d:=DerivedSubgroup(G);
+  ind:=IndexNC(G,d);
+  if ind=1 then return G;
+  # derived index small enough
+  elif ind<=limit then return d;
+  elif IsPrimeInt(ind) then return d;fi;
+
+  # make a p group
+  ind:=List(Collected(Factors(ind)),x->x[1]^x[2]);
+  Sort(ind);
+  if Length(ind)>1 then 
+    i:=0;
+    while i<Length(ind) and ind[i+1]<=limit do
+      i:=i+1;
+    od;
+    ind:=List(ind{Difference([1..Length(ind)],[i])},x->Factors(x)[1]);
```

Comment:
I'm taking the largest prime power dividing $[G:G']$ that is not larger than limit (since then I can factor out all other Sylow subgroups and get a characteristic subgroup of small index).
Yes, your code would be equivalent.